cmd/collector/app/span_processor.go

```diff
@@ -180,23 +182,55 @@ func (sp *spanProcessor) processItemFromQueue(item *queueItem) {
 	sp.metrics.InQueueLatency.Record(time.Since(item.queuedTime))
 }
 
-func (sp *spanProcessor) addCollectorTags(span *model.Span) {
-	if len(sp.collectorTags) == 0 {
-		return
-	}
-	dedupKey := make(map[string]struct{})
-	for _, tag := range span.Process.Tags {
-		if value, ok := sp.collectorTags[tag.Key]; ok && value == tag.AsString() {
-			sp.logger.Debug("ignore collector process tags", zap.String("key", tag.Key), zap.String("value", value))
-			dedupKey[tag.Key] = struct{}{}
+func (sp *spanProcessor) deDupTags(span *model.Span) {
+	deDupKey := make(map[string]int, len(span.Process.Tags))
+	deDupKeyVal := make(map[string]int, len(span.Process.Tags))
+	for idx, tag := range span.Process.Tags {
```

Comment:
`idx` isn't used after setting them in the respective maps. I suggest changing the maps to be `map[string]bool`, and set them to `true`.

Then you could do `if !dedupKey[k] {`

Reply:
`idx` is being used in `replace` case for deleting the tag from array.